### PR TITLE
#164798220 Highlight and Comment on texts in Articles 

### PR DESCRIPTION
**Pull request by dorothyas**
**What does this PR do?**

- Enables users to highlight and comment on an article

**Description of Task to be completed?**

**How should this be manually tested?**

- Fetch this branch git fetch origin `ft-highlight-comment-article-164798220`
- Checkout to the branch git checkout `ft-highlight-comment-article-164798220`
- Run `python manage.py makemigrations`
- Run `python manage.py migrate`
- Run `python manage.py runserver`
- Login a user POST /api/users/login/
- Create an article POST /api/articles/
- Create a comment POST /api/articles/<str:slug>/comments/
the format of your comment should include the area of the article to highlight as below:

```{
  "comment": {
    "body": "i want to comment on this article",
    "start_position":6,
     "end_position":12
  }
}
```

 **What are the relevant pivotal tracker stories?**

[#164798220](https://www.pivotaltracker.com/n/projects/2320796/stories/164798220)

**Screenshots**

Screenshot 2019-04-17 at 16 52 27
The start_position and end position show the area of the article that has been highlighted

### CODE REVIEW

**Review by felixkiryowa — COMMENTED** [review]
@dorothyas 
Thanks for the good work, I would request you remove the spaces you added to the articles views file because  it is not necessary
Screenshot 2019-04-18 at 05 46 25

**Comment by dorothyas** [comment]
> @dorothyas
> Thanks for the good work, I would request you remove the spaces you added to the articles views file because it is not necessary
> Screenshot 2019-04-18 at 05 46 25

I did not add any space. I just re-arranged the the code

**Comment by felixkiryowa** [comment]
> > @dorothyas
> > Thanks for the good work, I would request you remove the spaces you added to the articles views file because it is not necessary
> > Screenshot 2019-04-18 at 05 46 25
> 
> I did not add any space. I just re-arranged the the code

> when you check in file changes in both files that's how it looked like

File: authors/apps/articles/migrations/0001_initial.py [hunk 1]
```diff
@@ -1,4 +1,4 @@
-# Generated by Django 2.2 on 2019-04-11 05:41
+# Generated by Django 2.2 on 2019-04-17 08:45
```
Comment (ja-odur) [inline authors/apps/articles/migrations/0001_initial.py]: @dorothyas please dont merge/combine migrations

File: authors/apps/articles/migrations/0002_auto_20190411_1357.py [hunk 2]
```diff
@@ -1,35 +0,0 @@
-# Generated by Django 2.2 on 2019-04-11 13:57
```
Comment (ja-odur) [inline authors/apps/articles/migrations/0002_auto_20190411_1357.py]: @dorothyas please dont delete existing migrations

File: authors/apps/articles/migrations/0003_article_favorite.py [hunk 3]
```diff
@@ -1,20 +0,0 @@
-# Generated by Django 2.2 on 2019-04-11 15:28
-
-from django.conf import settings
-from django.db import migrations, models
```
Comment (ja-odur) [inline authors/apps/articles/migrations/0003_article_favorite.py]: @dorothyas same here dont delete existing migrations

File: authors/apps/authentication/migrations/0002_auto_20190411_0532.py [hunk 4]
```diff
@@ -1,23 +0,0 @@
-# Generated by Django 2.2 on 2019-04-11 05:32
-
-from django.db import migrations, models
```
Comment (ja-odur) [inline authors/apps/authentication/migrations/0002_auto_20190411_0532.py]: same here

File: authors/apps/profiles/migrations/0001_initial.py [hunk 5]
```diff
@@ -1,4 +1,4 @@
-# Generated by Django 2.2 on 2019-04-02 18:04
```
Comment (ja-odur) [inline authors/apps/profiles/migrations/0001_initial.py:1]: @dorothyas undo this as well

**Review by ja-odur — CHANGES_REQUESTED** [review]

**Pull request merged**